src/Common/src/System/Security/Cryptography/Oids.cs

```diff
@@ -26,6 +26,7 @@ internal static partial class Oids
         internal const string RsaPkcs1Sha512 = "1.2.840.113549.1.1.13";
         internal const string Esdh = "1.2.840.113549.1.9.16.3.5";
         internal const string EcDiffieHellman = "1.3.132.1.12";
+        internal const string X942dh = "1.2.840.10046.2.1";
```

Comment:
`DiffieHellman` or `DiffieHellmanPublicKey` seems better.  (https://tools.ietf.org/html/rfc3370#section-4.1.1 calls it `dh-public-number`, which is weird, don't they know they generally start with `id-`? :smile:)